bin/stress-test/src/main.rs

```diff
@@ -0,0 +1,377 @@
+
+        consume_notes_txs = accounts_and_notes;
+
+        // Create the block and send it to the store
+        let (elapsed, block_size) = apply_block(batches.clone(), store_client).await;
+
+        // Track metrics
+        metrics.add_insertion(elapsed, block_size);
+        if i % 50 == 0 {
+            metrics.add_store_size(get_store_size(dump_file));
+        }
+    }
+    metrics
+}
+
+/// Given a list of batches, create a `ProvenBlock` and send it to the store.
+/// Returns a tuple with:
+/// - the time spent on executing `StoreClient::apply_block`
+/// - the size of the block in bytes
+async fn apply_block(batches: Vec<ProvenBatch>, store_client: &StoreClient) -> (Duration, usize) {
+    let inputs = store_client
+        .get_block_inputs(
+            batches.iter().flat_map(ProvenBatch::updated_accounts),
+            batches.iter().flat_map(ProvenBatch::created_nullifiers),
+            batches.iter().flat_map(|batch| {
+                batch
+                    .input_notes()
+                    .iter()
+                    .cloned()
+                    .filter_map(|note| note.header().map(NoteHeader::id))
+            }),
+            batches.iter().map(ProvenBatch::reference_block_num),
+        )
+        .await
+        .unwrap();
+    let proposed_block = ProposedBlock::new(inputs, batches).unwrap();
+    let proven_block = LocalBlockProver::new(0)
+        .prove_without_batch_verification(proposed_block)
+        .unwrap();
+
+    let start = Instant::now();
+    let block_size: usize = proven_block.to_bytes().len();
+    store_client.apply_block(&proven_block).await.unwrap();
+    (start.elapsed(), block_size)
+}
+
+// HELPERS
+// -------------------------------------------------------------------------------------------------
+
+/// Create accounts and notes that mint assets for a given number of accounts.
+///
+/// Returns a vector of tuples, where each tuple contains:
+/// - the new account id
+/// - the new note that sends assets to the account
+/// - the proven transaction that consumes the note
+fn create_accounts_and_notes(
+    genesis_header: &BlockHeader,
+    key_pair: &SecretKey,
+    rng: &Arc<Mutex<RpoRandomCoin>>,
+    faucet_id: AccountId,
+    num_accounts: usize,
+) -> Vec<(AccountId, Note, ProvenTransaction)> {
+    (0..num_accounts)
+        .into_par_iter()
+        .map(|account_index| {
+            let account =
+                create_account(genesis_header, key_pair.public_key(), account_index as u64);
+            let note = {
+                let mut rng = rng.lock().unwrap();
+                create_note(faucet_id, account, &mut rng)
+            };
+
+            let path = MerklePath::new(vec![]);
+            let inclusion_proof = NoteInclusionProof::new(0.into(), 0, path).unwrap();
+
+            let consume_tx = create_tx(
+                account,
+                Digest::default(),
+                Digest::default(),
```

Comment:
It may be good to use actual initial/final hashes here - this way, the database would be more correct. To do this, we would need to return the actual account from `create_account()`. Also, it may make sense to move the account update logic into `create_tx()` function. It could look something like this:
```rust
fn create_consume_note_tx(
    account: Account,
    input_note: InputNote,
) -> ProvenTransaction
```
This would also mean that we'd need a different function for creating the transaction which emits nots from the faucet. It could look something like:
```rust
fn create_emit_notes_tx(
    faucet: Account,
    output_notes: Vec<Note>,
) -> ProvenTransaction
```